discretize/utils/meshutils.py

```diff
@@ -770,3 +771,113 @@ def inBox(cell):
         )
 
     return mesh
+
+
+def active_from_xyz(mesh, xyz, grid_reference='CC', method='linear'):
+    """Returns an active cell index array below a surface
+
+    Get active cells in the `mesh` that are below the surface create by
+    interpolating over the last dimension of the input points. This method will
+    uses scipy's interpolation routine to interpolate between input points. This
+    will use a nearest neighbour interpolation for cell values outside the convex
+    hull of points.
+
+    For `grid_reference='CC'`, this will check if the center of a given cell in
+    the mesh is below the interpolation surface to determine if that cell is
+    active.
+
+    For `grid_reference='N'`, this will check if **every** node of a given cell
+    in the mesh is below the interpolation surface.
+
+    For the
+
+    Parameters
+    ----------
+
+    mesh : TensorMesh or TreeMesh
+        Mesh object
+    xyz : numpy.ndarray
+        Points coordinates shape (*, mesh.dim).
+    grid_reference : {'CC', 'N'}
+        Use cell coordinates from cells-center 'CC' or nodes 'N'.
+    method : {'linear', 'nearest'}
+        Interpolation method for the xyz points.
+
+    Returns
+    -------
+
+    active : numpy.ndarray
+        1D mask array of `bool` for the active cells below xyz.
+    """
+    if not isinstance(mesh, (discretize.TensorMesh, discretize.TreeMesh)):
+        raise TypeError("Mesh must be either TensorMesh or TreeMesh")
+
+    if grid_reference not in ["N", "CC"]:
+        raise ValueError("Value of grid_reference must be 'N' (nodal) or 'CC' (cell center)")
+
+    dim = mesh.dim - 1
+
+    if mesh.dim == 3:
```

Comment:
here, this won't work as is for a fully 3D cyl - mesh (because we use the `y`-dimension to capture `theta`). So we would need to instead grab the associated `cartesianGrid` (https://github.com/simpeg/discretize/blob/master/discretize/CylMesh.py#L1607-L1625) before any interpolation. 

For now, I would recommend just doing a quick check and raise a `NotImplementedError`

```
if isinstance(mesh, discretize.CylMesh) and not mesh.isSymmetric:
    raise NotImplementedError
```

Reply:
(it will be okay for the cylindrically symmetric meshes because there is no variation in y - which is what the test confirms)